turtle_tf2_cpp/src/turtle_tf2_listener.cpp

```diff
@@ -88,42 +66,74 @@ class FrameListener : public rclcpp::Node
     std::string fromFrameRel = target_frame_.c_str();
     std::string toFrameRel = "turtle2";
 
-    geometry_msgs::msg::TransformStamped transformStamped;
-
-    // Look up for the transformation between target_frame and turtle2 frames
-    // and send velocity commands for turtle2 to reach target_frame
-    try {
-      transformStamped = tf_buffer_->lookupTransform(
-        toFrameRel, fromFrameRel,
-        tf2::TimePoint(),
-        500ms);
-    } catch (tf2::LookupException & ex) {
-      RCLCPP_INFO(this->get_logger(), "transform not ready");
+    if (!service_called_) {
+      // Check if the service is ready
+      if (!spawner_->service_is_ready()) {
+        RCLCPP_INFO(this->get_logger(), "Service is not ready");
+        return;
+      }
+
+      // Initialize request with turtle name and coordinates
+      // Note that x, y and theta are defined as floats in turtlesim/srv/Spawn
+      auto request = std::make_shared<turtlesim::srv::Spawn::Request>();
+      request->x = 4.0;
+      request->y = 2.0;
+      request->theta = 0.0;
+      request->name = "turtle2";
+      // Call request
+      result_ = spawner_->async_send_request(request);
+      service_called_ = true;
       return;
+    } else if (service_called_ && !turtle_spawned_) {
```

Comment:
you are assuming here that the service was called and answered.

take a look to this [example](https://github.com/ros2/demos/blob/master/demo_nodes_cpp/src/services/add_two_ints_client_async.cpp)

you should create the lambda or method to get the response. In our case set to `true`a boolean.

Reply:
Is it the same if I just call the `result_.get()` and see if it is true?

Reply:
`get()` returns the result of the service, but there is no garanties that you will get the response only calling `get()`, you might get a wrong answer. For this reason you use the callback.

Reply:
It makes code much more complex. I can reimplement it using the classical `send_request()`. What do you think about it? I will also need to change it in the Python example